nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviService.java

```diff
@@ -113,17 +113,36 @@ private static List<ApprovalStatus> generatePendingApprovalStatuses(Set<URI> ins
                    .toList();
     }
 
-    private static Candidate toPendingCandidate(CandidateDetails candidateDetails,
+    @JacocoGenerated
+    @SuppressWarnings("PMD.UnusedPrivateMethod")
+    private boolean exists(UUID uuid) {
+        return nviCandidateRepository.findById(uuid).isPresent();
+    }
+
+    @JacocoGenerated //TODO: Remove when its actually used
+    public Optional<CandidateWithIdentifier> findById(UUID uuid) {
+        return nviCandidateRepository.findById(uuid);
+    }
+
+    @JacocoGenerated //TODO: Remove when its actually used
+    public Optional<CandidateWithIdentifier> findByPublicationId(URI publicationId) {
+        return nviCandidateRepository.findByPublicationId(publicationId);
+    }
+
+    private static Candidate toPendingCandidate(CandidateEvaluatedMessage candidateEvaluatedMessage,
                                                 Map<URI, BigDecimal> institutionPoints) {
-        return Candidate.builder()
-                   .withPublicationId(candidateDetails.publicationId())
+        return new Candidate.Builder()
+                   .withPublicationBucketUri(candidateEvaluatedMessage.publicationBucketUri())
+                   .withPublicationId(candidateEvaluatedMessage.candidateDetails().publicationId())
                    .withIsApplicable(true)
-                   .withCreators(mapToVerifiedCreators(candidateDetails.verifiedCreators()))
-                   .withLevel(Level.parse(candidateDetails.level()))
-                   .withInstanceType(candidateDetails.instanceType())
-                   .withPublicationDate(mapToPublicationDate(candidateDetails.publicationDate()))
+                   .withCreators(mapToVerifiedCreators(candidateEvaluatedMessage.candidateDetails().verifiedCreators()))
+                   .withLevel(Level.parse(candidateEvaluatedMessage.candidateDetails().level()))
+                   .withInstanceType(candidateEvaluatedMessage.candidateDetails().instanceType())
+                   .withPublicationDate(mapToPublicationDate(candidateEvaluatedMessage.candidateDetails()
+                                                                 .publicationDate()))
                    .withPoints(institutionPoints)
-                   .withApprovalStatuses(generatePendingApprovalStatuses(institutionPoints.keySet()))
+                   .withApprovalStatuses(generatePendingApprovalStatuses(
```

Comment:
I can see you've had merge conflicts here, this is the newest code for `.withApprovalStatuses`:
`.withApprovalStatuses(generatePendingApprovalStatuses(institutionPoints.keySet()))`